processor/configuration/src/main/java/com/redhat/service/smartevents/processor/GatewayConfigurator.java

```diff
@@ -0,0 +1,66 @@
+package com.redhat.service.smartevents.processor;
+
+import java.util.Optional;
+
+import com.redhat.service.smartevents.infra.exceptions.definitions.user.GatewayProviderException;
+import com.redhat.service.smartevents.infra.models.gateways.Action;
+import com.redhat.service.smartevents.infra.models.gateways.Source;
+
+public interface GatewayConfigurator {
```

Comment:
I think `actionType` and `sourceType` should be enums to clearly define what values are supported/allowed.

Maybe that is 'good-first-issue' for me that I could work on in parallel in separate PR?

Reply:
Yes we can think about this after Event Sources are implemented :+1:

Reply:
Created https://issues.redhat.com/browse/MGDOBR-632